Formula/llvm.rb

```diff
@@ -108,33 +128,54 @@ class Llvm < Formula
   keg_only :provided_by_osx
 
   option :universal
-  option "with-clang", "Build the Clang compiler and support libraries"
-  option "with-clang-extra-tools", "Build extra tools for Clang"
-  option "with-compiler-rt", "Build Clang runtime support libraries for code sanitizers, builtins, and profiling"
-  option "with-libcxx", "Build the libc++ standard library"
-  option "with-lld", "Build LLD linker"
+  option "without-clang", "Do not build the Clang compiler and support libraries"
+  option "without-clang-extra-tools", "Do not build extra tools for Clang"
+  option "without-compiler-rt", "Do not build Clang runtime support libraries for code sanitizers, builtins, and profiling"
+  option "without-libcxx", "Do not build the libc++ standard library"
+  option "with-libcxxabi", "Build the libc++abi standard library"
+  option "with-libunwind", "Build the libunwind library"
+  option "without-lld", "Do not build LLD linker"
   option "with-lldb", "Build LLDB debugger"
+  option "with-openmp", "Build additional OpenMP Runtime Libraries"
   option "with-python", "Build Python bindings against Homebrew Python"
-  option "with-rtti", "Build with C++ RTTI"
-  option "with-utils", "Install utility binaries"
-
-  deprecated_option "rtti" => "with-rtti"
-
-  if MacOS.version <= :snow_leopard
+  option "without-rtti", "Build without C++ RTTI"
+  option "without-utils", "Do not install utility binaries"
+  option "without-polly", "Build without Polly optimizer"
+  option "with-test", "Build LLVM unit tests"
+  option "with-shared-libs", "Build all libs as shared instead of static (for developers)"
```

Comment:
I'm fine with making the text less specific &mdash; I just want that users understand what this option does. By default, llvm builds individual libraries as static. If option `DLLVM_BUILD_LLVM_DYLIB` is turned on (and it is turned on by default in the current formula), then these libraries are dynamically linked into a single shared library `libLLVM`. This option results in all individual libraries being built as shared.
Now, that I looked closer at this, there is the following note at LLVM website:

> LLVM_BUILD_LLVM_DYLIB:BOOL
> If enabled, the target for building the libLLVM shared library is added. This library contains all of LLVM’s components in a single shared library. Defaults to OFF. **This cannot be used in conjunction with BUILD_SHARED_LIBS**. Tools will only be linked to the libLLVM shared library if LLVM_LINK_LLVM_DYLIB is also ON. The components in the library can be customised by setting LLVM_DYLIB_COMPONENTS to a list of the desired components.

So, I have to modify the formula accordingly